assets/js/payment-method-extensions/payment-methods/stripe/payment-request/payment-request-express.js

```diff
@@ -0,0 +1,380 @@
+/**
+ * Internal dependencies
+ */
+import {
+	DEFAULT_STRIPE_EVENT_HANDLERS,
+	PAYMENT_METHOD_NAME,
+} from './constants';
+import {
+	getStripeServerData,
+	getPaymentRequest,
+	updatePaymentRequest,
+	canDoPaymentRequest,
+	getTotalPaymentItem,
+	getBillingData,
+	getPaymentMethodData,
+	getShippingData,
+	normalizeShippingAddressForCheckout,
+	normalizeShippingOptions,
+	normalizeLineItems,
+	normalizeShippingOptionSelectionsForCheckout,
+} from '../stripe-utils';
+
+/**
+ * External dependencies
+ */
+import { useRef, useState, useEffect } from '@wordpress/element';
+import {
+	Elements,
+	PaymentRequestButtonElement,
+	useStripe,
+} from '@stripe/react-stripe-js';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * @typedef {import('../stripe-utils/type-defs').Stripe} Stripe
+ * @typedef {import('../stripe-utils/type-defs').StripePaymentRequest} StripePaymentRequest
+ * @typedef {import('@woocommerce/type-defs/registered-payment-method-props').RegisteredPaymentMethodProps} RegisteredPaymentMethodProps
+ */
+
+/**
+ * @typedef {Object} WithStripe
+ *
+ * @property {Stripe} [stripe] Stripe api (might not be present)
+ */
+
+/**
+ * @typedef {RegisteredPaymentMethodProps & WithStripe} StripeRegisteredPaymentMethodProps
+ */
+
+/**
+ * PaymentRequestExpressComponent
+ *
+ * @param {StripeRegisteredPaymentMethodProps} props Incoming props
+ */
+const PaymentRequestExpressComponent = ( {
+	shippingData,
+	billing,
+	eventRegistration,
+	onSubmit,
+	activePaymentMethod,
+	setActivePaymentMethod,
+	setExpressPaymentError,
+} ) => {
+	/**
+	 * @type {[ StripePaymentRequest|null, function( StripePaymentRequest ):StripePaymentRequest|null]}
+	 */
+	// @ts-ignore
+	const [ paymentRequest, setPaymentRequest ] = useState( null );
+	const stripe = useStripe();
+	const [ canMakePayment, setCanMakePayment ] = useState( false );
+	const [ paymentRequestType, setPaymentRequestType ] = useState( '' );
+	const [ isProcessing, setIsProcessing ] = useState( false );
+	const [ isFinished, setIsFinished ] = useState( false );
+	const eventHandlers = useRef( DEFAULT_STRIPE_EVENT_HANDLERS );
+	const currentBilling = useRef( billing );
+	const currentShipping = useRef( shippingData );
+	const currentPaymentRequest = useRef( paymentRequest );
+	// for keeping track of what the active payment method was before this
+	// payment button was clicked.
+	const originalActivePaymentMethod = useRef( activePaymentMethod );
```

Comment:
Struggling to understand why this is needed.

Reply:
When the checkout form is loaded, the active payment method is whatever is set for the payment methods tab embedded in the checkout. However, when express payment method button is clicked, it needs to set the active payment method to itself so checkout knows what payment method is active. The original active payment method needs to be kept so that when the express payment interface is closed by the express payment method, the active payment method can be reverted back to what it was in checkout (which is important if shopper changes their mind and doesn't want to - or can't - pay with the express payment method).

Ideally though, this logic would be handled by checkout so I'm going to add that to followup.